packages/discovery-react-components/src/components/ResultsPagination/ResultsPagination.tsx

```diff
@@ -63,7 +88,49 @@ const ResultsPagination: FC<ResultsPaginationProps> = ({
     isResultsPaginationComponentHidden
   } = useContext(SearchContext);
   const [currentPage, setCurrentPage] = useState(page);
-  const resultsPerPage = componentSettings?.results_per_page ?? 10;
+  const [resetCounter, setResetCounter] = useState(0);
+  const resultsPerPage = pageSize || searchParameters.count || 10;
+
+  const handleOnChange = useCallback(
+    (evt: ResultsPaginationEvent): void => {
+      if (onChange) {
+        onChange(evt);
+      }
+      const { page, pageSize } = evt;
+      const offset = (page - 1) * pageSize;
+      setCurrentPage(page);
+      performSearch(
+        {
+          ...searchParameters,
+          count: pageSize,
+          offset
+        },
+        false
+      );
+    },
+    [onChange, performSearch, searchParameters]
+  );
+
+  // Since the Carbon pagination component controls the page number via its state
+  // without a way to influence it via props, we occasionally need to reset it back to "Page 1".
+  // This is done via a key on the CarbonPagination element, which is triggered via a counter
+  // that is incremented by this function.
+  const reset = useCallback(
+    ({ triggerOnChange }: ResetOptions) => {
```

Comment:
Will a consumer of this component ever want to call `reset()` with `triggerOnChange: true`? Or is this just for internal usage only? If the latter, then don't expose it as part of the external API.

Reply:
Yes, we will need to use it in `discovery-tooling`. :+1: